docs/frontend-system/utility-apis/04-configuring.md

```diff
@@ -8,74 +8,67 @@ description: Configuring, extending, and overriding utility APIs
-    }),
-  ),
-  /* highlight-add-end */
-  /* highlight-remove-next-line */
-  factory: createApiFactory({
-  /* highlight-add-next-line */
-  factory: ({ config }) => createApiFactory({
-    api: workApiRef,
-    deps: { storageApi: storageApiRef },
-    factory: ({ storageApi }) => {
-      /* highlight-add-start */
-      if (config.goSlow) {
-        /* ... */
-      }
-      /* highlight-add-end */
-    },
-  }),
-});
-```
+The config schema of the extension will tell you what configuration parameters it supports. Here we override the `goSlow` configuration value, which replaces the default.
 
-We wanted users to be able to configure a `goSlow` parameter for our API instances. So we added another interface type for holding our various options, and passed in a `configSchema` to `createApiExtension` which matches that interface. This example builds it using [the zod library](https://zod.dev/). The actual config values will then be passed in to the `factory` which is now a callback, wherein we can do what we wish with them. When changing to the callback form, we also had to add a top level `api: workApiRef` under `createApiExtension`.
+## Attaching extensions to inputs
 
-Note that while we use the word "config" here, it's _not_ the same thing as the `configApi` which gives you access to the full app-config. The config discussed here is instead the particular configuration settings given to your utility API instance. This is discussed more [in the section below](#configuring).
+Like with other extension types, you add input attachments to a Utility API by declaring the `attachTo` section of that attachment to point to the Utility APIs ID and input name.
 
-Note also that the config schema contained a default value fo the `goSlow` field. This is an important consideration. You want users of your API to be able to make maximum use of it, without having to dive deep into how to configure it. For that reason you generally want to provide as many sane defaults as possible, while letting users override them rarely but with purpose, only when called for. If you have a config schema without defaults, the framework will refuse to instantiate the utility API on startup unless the user had configured those values explicitly. Since it had a default value, the TypeScript code and interfaces also don't have to defensively allow `undefined` - we know that it'll have either the default value or an overridden value when we start consuming the config data.
+Well written input-enabled extension often have extension creator functions that help you make such attachments. Those functions typically set the `attachTo` section correctly on your behalf so that you don't have to figure them out.
 
-## Configuring
+## Replacing a Utility API implementation
 
-> TODO
+Like with other extension types, you replace Utility APIs with your own custom implementation using [extension overrides](../architecture/05-extension-overrides.md).
```

Comment:
Probably worth linking to this section from the override docs specifically for API extensions @camilaibs. I think it's important that this use-case is easily discoverable in there docs.

We could probably add a link from here to the override docs as well though, to handle the actual installation of the overrides and so on.

Might even be worth slimming down the example by skipping the installation and instead link to the override docs? Would have the benefit of more strongly encouraging a separate package.

Reply:
Aight let's do a stitching / cleanup step when all is merged and we have the full picture